Add documentation for adding a new track (#279)
exercism/docs: building/tracks/new/add-initial-exercises.md

| Lists                  | [resistor-color][resistor-color], [spiral-matrix][spiral-matrix]                     |
| Loops                  | [collatz-conjecture][collatz-conjecture], [protein-translation][protein-translation] |
| Maps                   | [nucleotide-count][nucleotide-count]                                                 |
| Numbers                | [leap][leap], [grains][grains]                                                       |
| Randomness             | [dnd-character][dnd-character], [robot-name][robot-name]                             |
| Strings                | [bob][bob], [isogram][isogram],                                                      |
| Trees                  | [binary-search-tree][binary-search-tree], [pov][pov]                                 |

### Select exercises with varying degrees of difficulty

Some exercises will be more difficult to implement than others.
You should aim to implement exercises with varying degrees of difficulty: some easy ones, some intermediate ones and some hard ones.

How difficult an exercise is depends on several factors, such as the involved concepts and the track's language.

> It might help to check what the exercise looks like in other tracks that have implemented the exercise, to better get feel for what the implementation could look like.

Reviewer: ```suggestion
It might help to check what the exercise looks like in other tracks that have implemented the exercise, to better get a feel for what the implementation could look like.
```

| [raindrops][raindrops]                   | Intermediate | Conditionals                          |
| [bob][bob]                               | Intermediate | Strings                               |
| [allergies][allergies]                   | Intermediate | Bit manipulation                      |
| [series][series]                         | Intermediate | Exceptions                            |
| [circular-buffer][circular-buffer]       | Intermediate | Classes                               |
| [meetup][meetup]                         | Intermediate | Dates and time                        |
| [yacht][yacht]                           | Intermediate | Lists                                 |
| [dominoes][dominoes]                     | Hard         | Tuples, Laziness                      |
| [diamond][diamond]                       | Hard         | String formatting                     |
| [alphametics][alphametics]               | Hard         | Laziness                              |
| [forth][forth]                           | Hard         | Parsing                               |

## Implement exercises

> Having selected the exercises you want include in your track, the next step is to implement them

Reviewer: ```suggestion
Having selected the exercises you want include in your track, the next step is to implement them.
```

Each of the above-mentioned exercises has three bits of information:

1. `description.md`: explains what the exercise is about (required)
2. `metadata.toml`: metadata like the exercise's blurb (required)
3. `canonical-data.json`: a set of input/output combinations that describe the behavior of the exercise (optional)

There are two options when implementing one of the above exercises:

> 1. Implement the exercise from scratch, using the test cases in the `canonical-data.json` file

Reviewer: ```suggestion
1. Implement the exercise from scratch, using the test cases in the `canonical-data.json` file.
```

This is a full sentence on its own regardless of it being a bullet point.

> 2. Port an implementation of the exercise from another track

Reviewer: ```suggestion
2. Port an implementation of the exercise from another track.
```

Same here.

The second option can be particularly appealing, as it can give you results quickly.

> Keep in mind though that you should tweak the implementation to best fit your track.

Reviewer: ```suggestion
Keep in mind, though, that you should tweak the implementation to best fit your track.
```

> As an example, some tracks don't use classes but only work with functions.

Reviewer: ```suggestion
As an example, some tracks do not use classes but only work with functions.
```
